wallet-test/src/test/scala/org/bitcoins/wallet/util/BitcoinSWalletTest.scala

```diff
@@ -0,0 +1,40 @@
+package org.bitcoins.wallet.util
+
+import org.bitcoins.core.config.RegTest
+import org.bitcoins.core.protocol.blockchain.ChainParams
+import org.bitcoins.core.util.BitcoinSLogger
+import org.bitcoins.db.UnitTestDbConfig
+import org.bitcoins.wallet.config.WalletDbManagement
+import org.bitcoins.wallet.models.{AccountDAO, AddressDAO, MnemonicCodeDAO}
+import org.scalatest.{AsyncFlatSpec, BeforeAndAfterAll}
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.concurrent.{Await, ExecutionContext}
+
+trait BitcoinSWalletTest
+    extends AsyncFlatSpec
+    with BeforeAndAfterAll
+    with BitcoinSLogger {
+  implicit val ec: ExecutionContext = ExecutionContext.Implicits.global
+
+  protected lazy val dbConfig: UnitTestDbConfig.type = UnitTestDbConfig
+  protected val chainParams: ChainParams = WalletTestUtil.chainParams
+
+  protected val addressDAO: AddressDAO = AddressDAO(dbConfig, chainParams)
+  protected val accountDAO: AccountDAO = AccountDAO(dbConfig, chainParams)
+  protected val mnemonicDAO: MnemonicCodeDAO = MnemonicCodeDAO(dbConfig)
+
+  /** Timeout for async operations */
+  protected val timeout: FiniteDuration = 10.seconds
+
+  protected val networkParam: RegTest.type = WalletTestUtil.networkParam
+
+  override protected def beforeAll(): Unit = {
+    Await.result(WalletDbManagement.dropAll(dbConfig), timeout)
+    Await.result(WalletDbManagement.createAll(dbConfig), timeout)
+  }
+
```

Comment:
I think we should create a `withWallet` test fixture similar to how i created fixtures in the chain project https://github.com/bitcoin-s/bitcoin-s-core/blob/node/chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala#L51

Reply:
Should the fixture pass in a 3-tuple of DAOs needed? That would cause DAO's that aren't used in the test to be passed in/created/dropped. Not necessarily an issue, but might be

Reply:
So I think what you should try for this fixutre business is having an ADT of case classes that wrap the fixtures that are needed and then use `FixtureTag`s to specify which test should receive which fixture (feel free to ping me about this)

Reply:
Basically `withFixture` should look at the tag and determine which fixture to provide, and then for the test you do a normal `in {` but then you have two cases rather than a normal lambda, one for the case class you expect, and then a catch all for anything else that leads to a fail case.